stages/org.osbuild.fips

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/python3
+"""
+Enables the system FIPS mode.
+
+This stage calls `fips-mode-setup` to enable system FIPS mode.
+
+Notes:
+  - Requires 'chroot' in the buildroot.
+  - Runs the 'fips-mode-setup' script from the image in the chroot.
+"""
+
+
+import os.path
+import subprocess
+import sys
+
+from osbuild import api
+from osbuild.util.mnt import MountGuard
+
+SCHEMA_2 = r"""
+"options": {
+  "additionalProperties": false,
+  "description": "Enables the system FIPS mode",
+  "properties": {
+    "boot_cfg": {
+      "type": "boolean",
+      "description": "Reconfigure the boot loader"
+    }
+  }
+}
+"""
+
+
+def main(tree, options):
+
+    boot_cfg = options.get("boot_cfg", False)
+
+    mounts = [
+        {"path": "/dev", "bind": True, "ro": True},
+        {"path": "/sys", "bind": True, "ro": True},
+        {"path": "/proc", "bind": True, "ro": False},
+    ]
+
+    with MountGuard() as mounter:
+        for mount in mounts:
+            src = mount["path"].lstrip("/")
+            dest = os.path.join(tree, src)
+            bind = mount["bind"]
+            ro = mount["ro"]
+            os.makedirs(dest, exist_ok=True)
+            os.chmod(dest, mount.get("mode", 0o755))
+            mounter.mount(src, dest, bind=bind, ro=ro)
```

Comment:
The `fips-mode-setup` command is run outside the MountGuard context which means the mounts are removed before the execution of the subprocess.

Reply:
Ouch!

Reply:
That's why it does not work :)

Reply:
I think it should work now